lib/doorkeeper/oauth/token_introspection.rb

```diff
@@ -160,19 +160,25 @@ def valid_token?
         @token&.accessible?
       end
 
+      def valid_authorized_token?
+        !authorized_token_matches_introspected? &&
+          authorized_token.accessible? &&
+          token_introspection_allowed?
+      end
+
       # RFC7662 Section 2.1
       def authorized_token_matches_introspected?
         authorized_token.token == @token&.token
       end
 
       # config constraints for introspection in Doorkeeper.configuration.allow_token_introspection
-      def token_introspection_allowed?(client)
+      def token_introspection_allowed?
```

Comment:
```ruby
# token_introspection.rb
def token_introspection_allowed?
  (..)
  allow_introspection.call(
    @token,
    authorized_client&.application,
    authorized_token
  )
end
```
This 👆 method is called at one of following two scenarios: when `authorized_token` or when `authorized_client`
1. when `authorized_token`: `authorized_client` should be `nil`
So it's _needless_ to call `allow_introspection` block with `authorized_client` method as parameter.
2. when `authorized_client`: `authorized_token` should be `nil`. The same as 👆

Should I pass `authorized_client` and `authorized_token` params clearly to this method, as below:
```ruby
# token_introspection.rb
def token_introspection_allowed?(client, auth_token) 
  (..)
  allow_introspection.call(@token, client, auth_token)
end
```
At (1), just call `token_introspection_allowed?(nil, authorized_token)`
At (2), `token_introspection_allowed?(authorized_client&.application, nil)`